src/job.rs

```diff
@@ -947,11 +1014,13 @@ fn suspend_smtp_thread(context: &Context, suspend: bool) {
 }
 
 fn send_mdn(context: &Context, msg_id: MsgId) -> Result<()> {
-    let msg = Message::load_from_db(context, msg_id)?;
-    let mimefactory = MimeFactory::from_mdn(context, &msg)?;
-    let rendered_msg = mimefactory.render()?;
-
-    add_smtp_job(context, Action::SendMdn, &rendered_msg)?;
+    job_add(
+        context,
+        Action::SendMdnToContact,
+        msg_id.to_u32() as i32,
```

Comment:
maybe use contact-id as foreign-id and the rfc-message-id as param (this way, there would be no need to query the message-table later at all)

in the job, we could query over all jobs with the same contact id, send them as one message by adding a SendMdn job, and then delete all jobs for the contact that are were really processed.